### PR TITLE
media-libs/libfishsound: EAPI7, improve ebuild

### PR DESCRIPTION
**Pull request by mm1ke**
Hi,

this adds a new revision of media-libs/libfishsound with EAPI7 and minor improvements.

diff -u:
```
--- libfishsound-1.0.0.ebuild   2017-09-11 19:58:55.249193523 +0200
+++ libfishsound-1.0.0-r1.ebuild        2018-06-13 21:39:23.897311217 +0200
@@ -1,16 +1,15 @@
-# Copyright 1999-2012 Gentoo Foundation
+# Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=2
-inherit eutils
+EAPI=7
 
 DESCRIPTION="Simple programming interface for decoding and encoding audio data using vorbis or speex"
-HOMEPAGE="http://www.xiph.org/fishsound/"
-SRC_URI="http://downloads.xiph.org/releases/libfishsound/${P}.tar.gz"
+HOMEPAGE="https://www.xiph.org/fishsound/"
+SRC_URI="https://downloads.xiph.org/releases/libfishsound/${P}.tar.gz"
 
 LICENSE="BSD"
 SLOT="0"
-KEYWORDS="amd64 x86"
+KEYWORDS="~amd64 ~x86"
 IUSE="flac speex"
 
 RDEPEND="media-libs/libvorbis
@@ -23,25 +22,25 @@
 # bug #395153
 RESTRICT="test"
 
+PATCHES=( "${FILESDIR}"/${P}-pc.patch )
+
 src_prepare() {
-       epatch "${FILESDIR}"/${P}-pc.patch
        sed -i \
                -e 's:doxygen:doxygen-dummy:' \
                configure || die
+
+       default
 }
 
 src_configure() {
-       local myconf=""
-       use flac || myconf="${myconf} --disable-flac"
-       use speex || myconf="${myconf} --disable-speex"
-
        econf \
                --disable-dependency-tracking \
-               ${myconf}
+               $(use_enable flac) \
+               $(use_enable speex)
 }
 
 src_install() {
        emake DESTDIR="${D}" \
-               docdir="${D}/usr/share/doc/${PF}" install || die
+               docdir="${D}/usr/share/doc/${PF}" install
        dodoc AUTHORS ChangeLog README
 }
```

### CODE REVIEW

File: media-libs/libfishsound/libfishsound-1.0.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		configure || die
+
+	default
+}
+
+src_configure() {
+	econf \
+		--disable-dependency-tracking \
+		$(use_enable flac) \
+		$(use_enable speex)
```
Comment (aballier) [inline media-libs/libfishsound/libfishsound-1.0.0-r1.ebuild]: are you sure the configure script is not buggy and --enable-speex/flac does not actually disable it ?

File: media-libs/libfishsound/libfishsound-1.0.0-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+src_install() {
+	emake DESTDIR="${D}" \
+		docdir="${D}/usr/share/doc/${PF}" install
+	dodoc AUTHORS ChangeLog README
```
Comment (a17r) [inline media-libs/libfishsound/libfishsound-1.0.0-r1.ebuild]: `einstalldocs`

**Comment by Zlogene** [comment]
@a17r  `einstalldocs` only installs a `CHANGELOG` file and not a `ChangeLog` one

**Comment by a17r** [comment]
Until you define `DOCS`

**Comment by Zlogene** [comment]
True, but it is the double work anyway

File: media-libs/libfishsound/libfishsound-1.0.0-r1.ebuild (hunk 1 above)
Comment (mm1ke) [reply media-libs/libfishsound/libfishsound-1.0.0-r1.ebuild]: You're right. The configure script is buggy. I've reverted  back to the previous use.

**Comment by mm1ke** [comment]
@a17r @Zlogene 
From the docs, actually both ```CHANGELOG``` and ```ChangeLog``` would be installed with the default behavior of ```einstalldocs```
However, the list also includes ```README*``` which would, in our case, install ```README.win32``` as well.
So, in any case, i would need to set the ```DOCS``` variable in order to keep the installed packages the same. I'm not against using ```einstalldocs``` but since i would need to set ```DOCS``` i though i'll save a line and just use ```dodoc```

**Comment by mm1ke** [comment]
just rebased this PR. Please let me know if i really need to update the ebuild for ```einstalldocs```

**Pull request closed without merging**